File: 프로그래머스/2/12909. 올바른 괄호/올바른 괄호.java
import java.util.*;
class Solution {
    boolean solution(String s) {
        boolean ans = true;
        Stack<Character> stack = new Stack<>();
        for (char e : s.toCharArray()) {
            if (e == '(') {
                stack.add(e);
            } else {
                if (stack.isEmpty() || stack.peek() != '(') {
                    ans = false;
                    break;
                } else {
                    stack.pop();
                }
            }
        }
        if (!stack.isEmpty()) ans = false;
        return ans;
    }
}

File: 프로그래머스/2/12909. 올바른 괄호/올바른 괄호.py
def solution(s):
    ans = True
    stack = []
    for e in s:
        if e == '(':
            stack.append(e)
        else:
            if not stack or stack[-1] != '(':
                ans = False
                break
            else: stack.pop()
    if stack: ans = False
    return ans